pkg/gofr/http/responder_test.go

```diff
@@ -86,9 +88,9 @@ func TestRespondWithApplicationJSON(t *testing.T) {
 		{"error response", nil, sampleError,
 			http.StatusNotFound, `{"error":{"message":"route not registered"}}`},
 		{"error response contains a nullable type with a nil value", newNilTemp(), sampleError,
-			http.StatusNotFound, `{"error":{"message":"route not registered"}}`},
+			http.StatusPartialContent, `{"error":{"message":"route not registered"}}`},
```

Comment:
This test case doesn't seem right! How are we having 206 when data is nil

Reply:
On [line 70 ](https://github.com/gofr-dev/gofr/pull/955/files#diff-c81cc86e5fa96339d98d0d0f499423f94fd66cb67d2c2c6903bcdb1d67c11fe7R60) use the `isNil()` function to check if data is nil or nullable type

Reply:
corrected